### ISSUE TITLE
Support for Freemarker.

### ISSUE THREAD

**Issue by Cocuyo** [Glavin001/atom-beautify#539]
Atom offers a good support for freemarker language. It could be nice if beautify could support freemarker also.

**Comment by prettydiff** [comment]
Freemarker appears to be a HTML template language so the beautification will likely be powered by the Pretty Diff tool.  In order for me to provide support I need to know more about the syntax.  From looking at the documentation I have gathered this:

    <#tag_name> 
    </#tag_name> 

Except, I am looking at documentation here: http://freemarker.org/docs/xgui_declarative_basics.html

`<#recurse doc>` does not appear to be a start tag. Maybe this is a self-closing tag.  If so could you provide me tag names for all self-closing tags, because this will drastically impact beautification.

**Comment by Cocuyo** [comment]
Hello, thanks for answering!

Besides the tags

<#tag_name> 
</#tag_name> 

There are also user defined directives:


<@tag_name> 
</@tag_name> 

The self closed directives are:

<#assign> (Which can also be explicitly closed)

<#break> (Always inside switch or list)

<#case> (Always inside switch)

<#default>

<#else> (Always inside if or list)

<#elseif> (Always inside if or list)

<#fallback>

<#flush>

<#ftl>

<#global> (Which can also be explicitly closed)

<#import>

<#include>

<#local>

<#t>

<#lt>

<#rt>

<#nested >

<#nt>

<#recover>

<#recurse>

<#return>

<#sep>

<#setting>

<#stop>

<#visit node>

**Comment by prettydiff** [comment]
Try it out at http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml?m=beautify

I have not yet added support for `#assign` or `#global` yet.  By * (Which can also be explicitly closed)* did you mean these tags may optionally come with a closing tag, like: `</#global>`?

**Comment by Cocuyo** [comment]
Yep, that's right. Those tags can be found in both forms (with closing tag
or without it).

**Comment by prettydiff** [comment]
All features should be supported at http://prettydiff.com/ignore/testlocation/prettydiff.com.xhtml?m=beautify

Just test it out and if there are any problems I will give it a quick fix.
